### DISCUSSION TITLE
Usage of nextclade CLI with Docker Image

### DISCUSSION THREAD

**Discussion by valentinelsra** [nextstrain/nextclade#551] [Help] [answered]
First Thanks for the development !

I need your help understand how can I use the docker image with my multi.fasta file... I tried to add the path with the argument --input-fasta but the file is not found.

Thanks ++,
Valentine.

**Comment by corneliusroemer** [comment, marked as answer]
Hi Valentine! I can try to help but I need more information. This is a kind of StackOverflow-like question you have so the same rules apply I guess, not sure whether you've ever posted a question there but this is the gist:
What are you trying to achieve? What's the command you've tried so far and what is the error message you get?

**Reply by valentinelsra** [reply]
Hello, first thank you to take some of your time answering me!

Actually I'm not used with StackOverflow.. So! The bio-informatician of the lab worked with nextflow and docker to adapt the artic pipeline for sars-cov-2 sequences analysis. I'm trying to keep the same strategy and use the docker image in order to add nextclade to this pipeline, but I'm really far from knowing how all of it works.

I first installed docker and then I pulled the image, downloaded the sars-cov-2 dataset and tried to run the analysis with example sequences.

The error message was a succession of this kind of message :
` [WARN] Nextclade: Warning: in sequence "USA/VA-DCLS-3814/2021": When processing gene "ORF6": Unable to align: no seed matches. Note that this gene will not be included in the results of the sequence. `

I don't know why there is no match between the example dataset and sequences. Do you know the reason? Anyway, I tried to run the analysis with my own sequences, inside a multi fasta, with the command below :
` nextclade \ --in-order \ --input-fasta=users/valentinelsra/test.multi.fasta \ --input-dataset=data/sars-cov-2\ --output-tsv=output/nextclade.tsv \ --output-tree=output/nextclade.auspice.json \ --output-dir=output/ \ --output-basename=nextclade `

The error message after that :
` [ERROR] Nextclade: Error: File not found: "users/valentinelsra/test.multi.fasta". Please verify correctness of input flags. If a dataset is used, check that the dataset is not corrupted. `

Is it clearer ?

Thanks ++,
Valentine

**Reply by corneliusroemer** [reply]
No worries, in that case go checkout StackOverflow, it's very useful ;)

Thanks for detailing your question. Now I can help much better.
>The error message was a succession of this kind of message :
`[WARN] Nextclade: Warning: in sequence "USA/VA-DCLS-3814/2021": When processing gene "ORF6": Unable to align: no seed matches. Note that this gene will not be included in the results of the sequence.`

This is in fact not an error at all. It's intended behaviour. Some sequences just cannot be aligned, usually because they are wrong. I can see how it's confusing if you don't know this. I will create an issue so that we can consider improving messaging of this warning. That it's an issue with the sequence, not with the tool. So actually all is working well.

Now regarding your command. The error message means that your file cannot be found/opened. Can you check that it does actually exist? What's the output of `head users/valinelsra/test.multi.fasta` when run in place of `nextclade ...` command?

I think what is actually happening is that this is a problem in the way you are using Docker. Have you used Docker before?

This should be your problem, you need to mount the directory to make the file available within Docker. I recommend you read up on Docker/mounting and/or ask your colleague who knows Docker. It should be very easy to solve your problem and make the data available from within Docker. In short: Docker is basically like a virtual machine, an operating system on its own. It does not have access to your filesystem, unless you provide that access explicitly. Read this or similar things: https://www.section.io/engineering-education/how-to-share-data-between-a-docker-container-and-the-host-computer/

I take your question as feedback, that we may want to make it more explicit, that you need to understand Docker to use the commands.

I hope the rest works out. Feel free to ask questions if you have any. I'll try my best to explain.

**Reply by valentinelsra** [reply]
Thank's a lot for your help and for taking into consideration my troubles in order to improve the user guide, it will help some other beginners like me. Actually you were right it was a problem related to my lack of knowledge concerning Docker. I have mounted the repertory and it finally worked! I've created a little bash script to run the nextclade image. However I can't understand the reason why, when I run this script with a cron, the docker command is not anymore found. Maybe you will have an explanation for it.

Thank's again,
Have a good day.

**Comment by corneliusroemer** [comment]
Glad to have been of help.

Cron runs as a different user (Linux) as you when you are in the terminal.
You need to set the path environment variable to include all the programs
you want to use.

I really recommend you try looking for answers at Stackoverflow. There's
like millions of questions and answers like yours ;)

https://serverfault.com/a/352838/544529

**Reply by valentinelsra** [reply]
Okay I will take a look on stack, thank's for the answer ! :)
